x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/observability_agent/tools/get_data_sources.spec.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { AGENT_BUILDER_ENABLED_SETTING_ID } from '@kbn/management-settings-ids';
+import type { ApmSynthtraceEsClient, LogsSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import { isOtherResult } from '@kbn/onechat-common/tools';
+import type { ToolResult, OtherResult } from '@kbn/onechat-common';
+import type { LlmProxy } from '@kbn/test-suites-xpack-platform/onechat_api_integration/utils/llm_proxy';
+import { createLlmProxy } from '@kbn/test-suites-xpack-platform/onechat_api_integration/utils/llm_proxy';
+import { OBSERVABILITY_GET_DATA_SOURCES_TOOL_ID } from '@kbn/observability-agent-plugin/server/tools';
+import { OBSERVABILITY_AGENT_ID } from '@kbn/observability-agent-plugin/server/agent/register_observability_agent';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../ftr_provider_context';
+import { createOneChatApiClient } from '../utils/one_chat_client';
+import { setupToolCallThenAnswer } from '../utils/llm_proxy/scenarios';
+import { createSyntheticLogsData, createSyntheticApmData } from '../utils/synthtrace_scenarios';
+import {
+  createLlmProxyActionConnector,
+  deleteActionConnector,
+} from '../utils/llm_proxy/action_connectors';
+
+const LLM_EXPOSED_TOOL_NAME_FOR_GET_DATA_SOURCES = 'observability_get_data_sources';
+const USER_PROMPT = 'Do I have any data sources? ';
+
+export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const log = getService('log');
+  const kibanaServer = getService('kibanaServer');
+  const roleScopedSupertest = getService('roleScopedSupertest');
+
+  describe(`tool: ${OBSERVABILITY_GET_DATA_SOURCES_TOOL_ID}`, function () {
+    // LLM Proxy is not yet supported in cloud environments
+    this.tags(['skipCloud']);
+
+    let llmProxy: LlmProxy;
+    let connectorId: string;
+    let oneChatApiClient: ReturnType<typeof createOneChatApiClient>;
+
+    describe('POST /api/agent_builder/converse', () => {
+      let toolResponseContent: { results: ToolResult[] };
+      let apmSynthtraceEsClient: ApmSynthtraceEsClient;
+      let logsSynthtraceEsClient: LogsSynthtraceEsClient;
+
+      before(async () => {
+        // Enable Agent Builder (can be removed when the agent builder is enabled by default)
+        await kibanaServer.uiSettings.update({
+          [AGENT_BUILDER_ENABLED_SETTING_ID]: true,
+        });
```

Comment:
what about specifying this in the test config similar to `feature_flags.overrides.observabilityAgent.enabled`? Something like `uiSettings.overrides['agentBuilder:enabled']: true`

Reply:
Like here https://github.com/elastic/kibana/blob/54983842acea7b4f24ce2bafeb111c60acdf67ee/x-pack/solutions/observability/test/functional_with_es_ssl/config.ts#L23